impl/src/test/java/org/ehcache/EhcacheFIFOEvictionTest.java

```diff
@@ -0,0 +1,112 @@
+import org.ehcache.config.Eviction;
+import org.ehcache.config.ResourcePoolsBuilder;
+import org.ehcache.config.units.EntryUnit;
+import org.ehcache.internal.TestTimeSource;
+import org.ehcache.internal.TimeSourceConfiguration;
+import org.hamcrest.core.Is;
+import org.junit.Test;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+
+/**
+ * @author rism
+ */
+public class EhcacheFIFOEvictionTest {
+
+  @Test
+  public void testFIFOEvictionForSmallSample() {
+    TestTimeSource timeSource = new TestTimeSource();
+
+    CacheConfiguration<String, String> cacheConfiguration = CacheConfigurationBuilder.newCacheConfigurationBuilder()
+        .usingEvictionPrioritizer(Eviction.Prioritizer.FIFO)
+        .add(new TimeSourceConfiguration(timeSource))
+        .withResourcePools(ResourcePoolsBuilder.newResourcePoolsBuilder()
+            .heap(5L, EntryUnit.ENTRIES).build()).buildConfig(String.class, String.class);
+
+    CacheManager cacheManager = CacheManagerBuilder.newCacheManagerBuilder().withCache("cache", cacheConfiguration)
+        .build(true);
+
+    Cache<String, String> cache = cacheManager.getCache("cache", String.class, String.class);
+
+//    Adding elements after time intervals
+    cache.put("1", "a");
+    timeSource.advanceTime(1);
+
+    cache.put("2", "b");
+    timeSource.advanceTime(1);
+
+    cache.put("3", "c");
+    timeSource.advanceTime(1);
+
+    cache.put("4", "d");
+    timeSource.advanceTime(1);
+
+    cache.put("5", "e");
+    timeSource.advanceTime(1);
+
+    cache.put("6", "f");
+
+//    asserting cache content
+    assertNull(cache.get("1"));
+    assertThat(cache.get("2"), is("b"));
+    assertThat(cache.get("3"), is("c"));
+    assertThat(cache.get("4"), is("d"));
+    assertThat(cache.get("5"), is("e"));
+    assertThat(cache.get("6"), is("f"));
+
+    cacheManager.close();
+  }
+
+  @Test
+  public void testFIFOEvictionForLargerSample() {
+    TestTimeSource timeSource = new TestTimeSource();
+
+    CacheConfiguration<Number, Number> cacheConfiguration = CacheConfigurationBuilder.newCacheConfigurationBuilder()
+        .usingEvictionPrioritizer(Eviction.Prioritizer.FIFO)
+        .add(new TimeSourceConfiguration(timeSource))
+        .withResourcePools(ResourcePoolsBuilder.newResourcePoolsBuilder()
+            .heap(100L, EntryUnit.ENTRIES).build()).buildConfig(Number.class, Number.class);
+
+    CacheManager cacheManager = CacheManagerBuilder.newCacheManagerBuilder().withCache("cache", cacheConfiguration)
+        .build(true);
+
+    Cache<Number, Number> cache = cacheManager.getCache("cache", Number.class, Number.class);
+
+//    Adding elements after time intervals
+    for(int i = 0; i < 100; i++) {
+      cache.put(i, i);
+      timeSource.advanceTime(1);
+    }
+
+    cache.put(100, 100);
+
+//    asserting last-in elements are never evicted
+    for(int i = 94; i < 101; i++) {
```

Comment:
I do not like the magic number used as lower bound here. Maybe we could have the sample size as a constant and use it here? And by changing this "integration" test to a "unit" test on `OnHeapStore`, it could probably remain package private.